sites/public/src/pages/listing/[id]/[slug].tsx

```diff
@@ -79,68 +80,27 @@ export default function ListingPage(props: ListingProps) {
     </Layout>
   )
 }
-/**
- *
- * getStaticPaths and getStaticProps with revalidation isn't actually working on netflify, so we have to use getServerSideProps until it does
- */
-/* export async function getStaticPaths(context: { locales: Array<string> }) {
-  try {
-    const response = await axios.get(process.env.listingServiceUrl, {
-      params: {
-        view: "base",
-        limit: "all",
-        filter: [
-          {
-            $comparison: "<>",
-            status: "pending",
-          },
-        ],
-      },
-      paramsSerializer: (params) => {
-        return qs.stringify(params)
-      },
-    })
 
-    return {
-      paths: response?.data?.items
-        ? context.locales.flatMap((locale: string) =>
-            response.data.items.map((listing) => ({
-              params: { id: listing.id, slug: listing.urlSlug },
-              locale: locale,
-            }))
-          )
-        : [],
-      fallback: "blocking",
-    }
-  } catch (error) {
-    console.error("listings getStaticPaths error = ", error)
-    return {
-      paths: [],
-      fallback: "blocking",
-    }
-  }
-} */
+export const getStaticPaths: GetStaticPaths = () => {
+  return { paths: [], fallback: "blocking" }
+}
 
-export async function getServerSideProps(context: {
+export const getStaticProps: GetStaticProps = async (context: {
   params: Record<string, string>
   locale: string
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  req: any
-}) {
+}) => {
   let response
   try {
     response = await axios.get(`${process.env.backendApiBase}/listings/${context.params.id}`, {
       headers: {
         language: context.locale,
         passkey: process.env.API_PASS_KEY,
-        "x-forwarded-for":
-          context.req.headers["x-forwarded-for"] ?? context.req.socket.remoteAddress,
       },
     })
   } catch (e) {
     return { notFound: true }
   }
-  const jurisdiction = fetchJurisdictionByName(context.req)
+  const jurisdiction = fetchJurisdictionByName()
 
-  return { props: { listing: response.data, jurisdiction: await jurisdiction } }
+  return { props: { listing: response.data, jurisdiction: await jurisdiction }, revalidate: 30 }
```

Comment:
suggestion: When a listing goes live or closes we want to make sure that this page is updated in a relatively quick amount of time so I think 30 seconds makes sense. But I could also see this being an environment variable that we could default to 30 seconds and override it in case we need to adjust.

Also, this doesn't have to be part of this PR, but Next just support [On-Demand ISR](https://nextjs.org/docs/14/pages/building-your-application/data-fetching/incremental-static-regeneration#on-demand-revalidation). So we could theoretically have this be a much larger "revalidate" time and have the backend refresh the page when the listing updates. That might be something we want to look at for long term sustainability

Reply:
I do like the idea of this eventually going to on-demand ISR!